java/code/src/com/suse/manager/webui/services/subscriptionmatching/SubscriptionMatchProcessor.java

```diff
@@ -163,6 +166,55 @@ private static String deriveMatchStatus(PinnedSubscription ps, InputJson input,
                  ));
     }
 
+    /**
+     * Merge subscriptions based on data in messages.
+     *
+     * @param subscriptions the subscriptions
+     * @param messages the messages
+     * @return
+     */
+    private List<SubscriptionJson> mergeSubscriptions(List<SubscriptionJson> subscriptions,
+            List<MessageJson> messages) {
+        Map<Long, SubscriptionJson> subsById = subscriptions.stream().collect(toMap(s -> s.getId(), s ->s));
+
+        // sub id -> sum of quantities of subs merged into it, latest start date/soonest end date over all merged subs
+        Map<Long, Tuple3<Integer, Date, Date>> mergedSubscriptionParams = messages.stream()
```

Comment:
Possible typo detected:
> Can't we just squash subscriptions and give Manager clean data to begin with?

Did you perhaps mean:
Can't we just squash subscriptions and give **Matcher** clean data to begin with?

Reply:
Ok, understood your point. I made the matcher output the "processed input" as well and I made SUMA use this instead of pass-through-input (see https://github.com/openSUSE/subscription-matcher/pull/12).